### PR TITLE
optionally use pkg-config (when present) to detect *_ROOTs

### PR DESCRIPTION
**Pull request by garbas**
This change is [Reviewable](https://reviewable.io/reviews/python-pillow/pillow/2074)

### CODE REVIEW

**Comment by garbas** [comment]
rebased on current master to re-trigger CI builds

**Comment by garbas** [comment]
everything builds ok, except pypi which is anyway broken in current master as well.

**Comment by garbas** [comment]
Is there anything else I can do?

**Comment by wiredfool** [comment]
Looking at the Travis logs, it appears that pkg-config is installed, but it's failing to find anything. I'm not seeing it do anything on my local machine either. 

How is this supposed to be setup? What environments is this intended to improve?

**Comment by radarhere** [comment]
Just a note that the master branch is now fixed, so if you rebase, all tests should pass.

**Comment by garbas** [comment]
just rebased ... lets see

**Comment by garbas** [comment]
@wiredfool ``pkg-config`` no idea how travis setup is configured. but ``pkg-config`` tool is the tool that Pillow should use anyway to look for headers of libraries. i believe that is the most prevalent way of looking for library headers.

In my case i'm using [Nix](https://nixos.org/nix) where header files are not located ``/usr/`` (actually there is no ``/usr``), but in some other locations exposed via system variables at build time so that tools like ``pkg-config`` can look at different places.

this patch just adds that by default it will use ``pkg-config`` (only if present) to look for header files. so you not seeing anything it is a good thing since everything should work as now, except on systems like ``nix`` being able to install Pillow is now possible.

and while writing this comment looks like all tests pass. yaaay!

File: setup.py [hunk 1]
```diff
@@ -119,6 +124,29 @@ def _read(file):
 FREETYPE_ROOT = None
 LCMS_ROOT = None
 
+# ask pkg-config
+if _cmd_exists('pkg-config'):
+    roots = dict(
+        JPEG_ROOT="libjpeg",
+        JPEG2K_ROOT="libopenjp2",
+        TIFF_ROOT="libtiff-4",
```
Comment (wiredfool) [inline setup.py]: We support both libtiff-4 (e.g. 3.9.x) and libtiff-5 (4.0.x)

**Comment by wiredfool** [comment]
When running `make clean`, I get the following output:
```
$  make clean
python setup.py clean
Package libtiff-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `libtiff-4.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libtiff-4' found
Package libtiff-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `libtiff-4.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libtiff-4' found
Package libimagequant was not found in the pkg-config search path.
Perhaps you should add the directory containing `libimagequant.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libimagequant' found
Package libimagequant was not found in the pkg-config search path.
Perhaps you should add the directory containing `libimagequant.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libimagequant' found
Package libjpeg was not found in the pkg-config search path.
Perhaps you should add the directory containing `libjpeg.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libjpeg' found
Package libjpeg was not found in the pkg-config search path.
Perhaps you should add the directory containing `libjpeg.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libjpeg' found
running clean
removing 'build/temp.linux-x86_64-2.7' (and everything under it)
rm PIL/*.so || true
rm: cannot remove `PIL/*.so': No such file or directory
rm -r build || true
find . -name __pycache__ | xargs rm -r || true
```

So it appears that you're running this any time that the setup runs.

File: setup.py (hunk 1 above)
Comment (garbas) [inline setup.py]: should i make it first search for ``libtiff-5`` and then ``libtiff-4``? or is ``libtiff-4`` preferred as a default?

**Comment by garbas** [comment]
@wiredfool tnx for taking the time for testing this. 

1. i will change that pkg-config is only called when build is triggered (not at every setup.py invocation)
2. i will hide all the debug output of pkg-config behind --debug flag
3. if pkg-config does not find libraries current steps should be taken and reported.

@wiredfool did i understand your comments correctly and is my todo list (above) complete? a :+1: is all i'm looking for before is start working on this ;)

**Comment by wiredfool** [comment]
Looks good. 

For debug reporting, I'd add success or failure for each item so it's clear where we're finding the libraries.

**Comment by aclark4life** [comment]
Weeeee, thanks @garbas ! Is this too risky for 3.4? If not, I say GO GO GO.

**Comment by wiredfool** [comment]
@aclark4life I was still waiting on the changes from my review.

**Comment by garbas** [comment]
@wiredfool @aclark4life rebased on master and fixed above mentioned issues.

**Comment by garbas** [comment]
i also squashed the commits

**Comment by wiredfool** [comment]
Merged in #2212

**Comment by garbas** [comment]
@wiredfool thank you!

**Pull request closed without merging**